src/node/validation_cache_args.cpp

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <node/validation_cache_args.h>
+
+#include <kernel/validation_cache_sizes.h>
+
+#include <util/system.h>
+
+#include <memory>
+#include <optional>
+
+using kernel::ValidationCacheSizes;
+
+namespace node {
+void ApplyArgsManOptions(const ArgsManager& argsman, ValidationCacheSizes& cache_sizes)
+{
+    if (auto max_size = argsman.GetIntArg("-maxsigcachesize")) {
+        cache_sizes = {
+            .signature_cache_bytes = *max_size / 2 * (1 << 20),
```

Comment:
In commit "validationcaches: Add and use ValidationCacheSizes" (e4d5bab88e9acad53cb8d636d836d2cdf9bb49b8)

I guess this would be a change in behavior, but maybe tweak order of operations so if a 1 MiB cache is requested this uses .5 MiB for each instead of 0

Reply:
Fixed as of eeb224e4d3, great catch!